packages/next/telemetry/events/plugins.ts

```diff
@@ -0,0 +1,186 @@
+import { record } from '../storage'
+import findUp from 'find-up'
+
+const EVENT_PLUGIN_PRESENT = 'NEXT_PLUGIN_PRESENT'
+type PluginPresentEventShape = {
+  pluginName: string
+  pluginVersion: string
+}
+
+const plugins = [
+  // CSS-in-JS solutions
```

Comment:
Out of curiosity I looked trough our package.json and while next-offline will be tracked - e.g. react-jss won't be as we implicitly use it trough material-ui.

Same is true for some other packages so I'm wondering if it would be better to scrape the lockfile instead?

As I side note: why this whitelisting at all? Wouldn't it be interesting to see if people don't use a certain plugin, but the vanilla packages instead to figure out what plugins to improve/promote?
Or to see which packages user use in combination to figure out what examples would be desired?

Reply:
@sakulstra the problem with that is that this solution has to be anonymous, obviously we'd love to track all deps if possible but that'd risk exposing internal package names, hence the whitelist.

@Timer had the same idea for using lockfiles, we'll use package-lock / yarn.lock instead after this lands 👍

Reply:
> but the vanilla packages instead to figure out what plugins to improve/promote?

What do you mean by vanilla packages? The built-in Next.js solution?

Reply:
> @sakulstra the problem with that is that this solution has to be anonymous, obviously we'd love to track all deps if possible but that'd risk exposing internal package names, hence the whitelist.

Wouldn't it be possible to somehow exclude private packages only :thinking:? Or allow users to opt-in to full package.json? While we have some private packages, I don't really care if you know that an <org>/ui-kit package exists :man_shrugging: 

> What do you mean by vanilla packages?
Packages which are used without a next plugin.

e.g. we're using `@apollo/...` but not `next-apollo` or similar, same for `i18next` and probably some other.
While for apollo the reason was simply that we didn't know this package exists :sweat_smile: (didn't yet check if it was useful for us)
For `next-18next` the reason for not using it was that it doesn't yet work with serverless.

I guess from a telemetry perspective it would be:
- interesting to see the size of apollo userbase within next
- interesting to see if a plugin usage dropped after a release (just a guess, but i think we're not the only ones who had to do quite some refactoring to get our app next 9 ready)

Reply:
@sakulstra yeah exactly, the list is not extensive enough right now (this is just the initial implementation).

@Timer let's add apollo and `i18next` / `react-intl` / `@lingui/react`